File: discord-bot/gpt-ignacio.js
require('dotenv').config();
const { Client, GatewayIntentBits, Partials } = require('discord.js');
const OpenAI = require('openai');

const client = new Client({
  intents: [
    GatewayIntentBits.Guilds,
    GatewayIntentBits.GuildMessages,
    GatewayIntentBits.MessageContent,
    GatewayIntentBits.DirectMessages,
  ],
  partials: [Partials.Channel],
});

const openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

client.once('clientReady', () => {
  console.log(`GPT_Ignacio conectado como ${client.user.tag}`);
});

client.on('messageCreate', async (msg) => {
  if (msg.author.bot) return;

  const isDM = msg.channel.type === 1;
  const isMention = msg.mentions.has(client.user);

  if (!isDM && !isMention) return;

  const texto = msg.content.replace(`<@${client.user.id}>`, '').trim();

  if (!texto) {
    return msg.reply('GPT_Ignacio listo. ¿Qué tarea debo ejecutar?');
  }

  if (texto === '!ping') {
    return msg.reply('GPT_Ignacio activo 🟢');
  }

  try {
    const completion = await openai.chat.completions.create({
      model: 'gpt-4o',
      messages: [
        {
          role: 'system',
          content: 'Eres GPT_Ignacio, un agente de IA de la Red de Agentes de Ignacio. Responde en español, de forma concisa y profesional.',
        },
        { role: 'user', content: texto },
      ],
    });

    const respuesta = completion.choices[0].message.content;
    await msg.reply(
      `Recibido. Soy GPT_Ignacio y ejecutaré la tarea asignada:\n"${texto}"\n\n${respuesta}`
    );
  } catch (err) {
    console.error('[GPT_Ignacio] Error OpenAI:', err.message);
    await msg.reply('Error al contactar con GPT. Reintenta en un momento.');
  }
});

client.login(process.env.GPT_IGNACIO_TOKEN);

File: bots/gpt_ignacio.py
"""
GPT_Ignacio — Discord bot powered by GPT-4o
"""
import os
import discord
from openai import OpenAI

DISCORD_TOKEN = os.environ["GPT_IGNACIO_TOKEN"]
OPENAI_API_KEY = os.environ["OPENAI_API_KEY"]

intents = discord.Intents.default()
intents.message_content = True
intents.members = True
intents.presences = True

client = discord.Client(intents=intents)
openai_client = OpenAI(api_key=OPENAI_API_KEY)

SYSTEM_PROMPT = (
    "Eres GPT_Ignacio, un agente IA en el servidor de Discord de Ignacio Concha. "
    "Formas parte de una red de agentes junto a Claude_Ignacio y ClawFrankie. "
    "Responde en español de manera concisa y útil. "
    "Cuando otro agente te hable, colabora con él."
)

@client.event
async def on_ready():
    print(f"GPT_Ignacio conectado como {client.user}")

@client.event
async def on_message(message):
    if message.author == client.user:
        return
    if client.user.mentioned_in(message) or isinstance(message.channel, discord.DMChannel):
        content = message.content.replace(f"<@{client.user.id}>", "").strip()
        if not content:
            return
        async with message.channel.typing():
            response = openai_client.chat.completions.create(
                model="gpt-4o",
                messages=[
                    {"role": "system", "content": SYSTEM_PROMPT},
                    {"role": "user", "content": content},
                ],
                max_tokens=1024,
            )
        await message.reply(response.choices[0].message.content)

client.run(DISCORD_TOKEN)
